### PR TITLE
Hello Hooks

### PR DESCRIPTION
**Pull request by hwillson**
This PR replaces `react-apollo` and the render prop components with `@apollo/react-hooks`. I've made a few additional changes at the same time, like updating project dependencies (excluding emotion) and fixing tests that were failing with recent versions of `apollo-client`.

### CODE REVIEW

File: final/client/src/pages/login.js [hunk 1]
```diff
@@ -11,26 +15,26 @@ export const LOGIN_USER = gql`
...
+    {
+      onCompleted({ login }) {
+        localStorage.setItem('token', login);
+        client.writeData({ data: { isLoggedIn: true } });
+      }
+    }
+  );
+
   return (
     <ApolloConsumer>
```
Comment (peggyrayzis) [inline final/client/src/pages/login.js]: Can we get rid of the `ApolloConsumer` component here now that we're using the `useApolloClient` hook?

File: final/client/src/containers/logout-button.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import React from 'react';
 import styled from 'react-emotion';
-import { ApolloConsumer } from 'react-apollo';
+import { ApolloConsumer } from '@apollo/react-hooks';
```
Comment (peggyrayzis) [inline final/client/src/containers/logout-button.js]: Switch to `useApolloClient` hook

File: final/client/src/pages/login.js (hunk 1 above)
Comment (hwillson) [reply final/client/src/pages/login.js]: Oops - definitely. Great catch!

File: final/client/src/containers/logout-button.js (hunk 2 above)
Comment (hwillson) [reply final/client/src/containers/logout-button.js]: Also a great point - thanks!

**Pull request merged**